crates/miden-testing/tests/agglayer/bridge_in.rs

```diff
@@ -0,0 +1,206 @@
+};
+use miden_objects::transaction::OutputNote;
+use miden_objects::{Felt, FieldElement, Word};
+use miden_testing::{AccountState, Auth, MockChain};
+use rand::Rng;
+
+/// Tests the bridge-in flow: CLAIM note -> Aggfaucet (FPI to Bridge) -> P2ID note created.
+#[tokio::test]
+async fn test_bridge_in_claim_to_p2id() -> anyhow::Result<()> {
+    let mut builder = MockChain::builder();
+
+    // CREATE BRIDGE ACCOUNT (with bridge_out component for MMR validation)
+    // --------------------------------------------------------------------------------------------
+    let bridge_storage_slot_name = StorageSlotName::new("miden::agglayer::bridge").unwrap();
+    let bridge_storage_slots = vec![StorageSlot::with_empty_map(bridge_storage_slot_name)];
+    let bridge_component = bridge_out_component(bridge_storage_slots);
+    let bridge_account_builder = Account::builder(builder.rng_mut().random())
+        .storage_mode(AccountStorageMode::Public)
+        .with_component(bridge_component);
+    let bridge_account = builder.add_account_from_builder(
+        Auth::IncrNonce,
+        bridge_account_builder,
+        AccountState::Exists,
+    )?;
+
+    // CREATE AGGLAYER FAUCET ACCOUNT (with agglayer_faucet component)
+    // --------------------------------------------------------------------------------------------
+
+    // Create network faucet storage slots (required for fungible asset creation)
+    let token_symbol = TokenSymbol::new("AGG").unwrap();
+    let decimals = 8u8;
+    let max_supply = Felt::new(1000000);
+
+    // Network faucet metadata slot: [max_supply, decimals, token_symbol, 0]
+    let metadata_word =
+        Word::new([max_supply, Felt::from(decimals), token_symbol.into(), Felt::ZERO]);
+    let metadata_slot =
+        StorageSlot::with_value(NetworkFungibleFaucet::metadata_slot().clone(), metadata_word);
+
+    // Agglayer-specific bridge storage slot
+    let bridge_account_id_word = Word::new([
+        Felt::new(0),
+        Felt::new(0),
+        bridge_account.id().suffix(),
+        bridge_account.id().prefix().as_felt(),
+    ]);
+    let agglayer_storage_slot_name = StorageSlotName::new("miden::agglayer::faucet").unwrap();
+    let bridge_slot = StorageSlot::with_value(agglayer_storage_slot_name, bridge_account_id_word);
+
+    // Combine all storage slots for the agglayer faucet component
+    let agglayer_storage_slots = vec![metadata_slot, bridge_slot];
+    let agglayer_component = agglayer_faucet_component(agglayer_storage_slots);
+
+    // Create agglayer faucet with FungibleFaucet account type and Network storage mode
+    let agglayer_faucet_seed = builder.rng_mut().random();
+    let agglayer_faucet_builder = Account::builder(agglayer_faucet_seed)
+        .account_type(AccountType::FungibleFaucet)
+        .storage_mode(AccountStorageMode::Network)  // Network faucets use Network storage mode
+        .with_component(agglayer_component);
+    let agglayer_faucet = builder.add_account_from_builder(
```

Comment:
same for the agglayer faucet: would be great to encapsulate the faucet creation logic in `pub fn create_agglayer_faucet()`. This helper would need to be aware of the agglayer bridge account id.